dask/dataframe/io/csv.py

```diff
@@ -941,7 +941,8 @@ def to_csv(
     if single_file:
         first_file = open_file(filename, mode=mode, **file_options)
         value = to_csv_chunk(dfs[0], first_file, **kwargs)
-        append_mode = mode.replace("w", "") + "a"
+        append_mode = mode if mode.endswith("a") else mode + "a"
```

Comment:
The mode should be order-independent, `at` would also be a valid mode (as mentioned in the docstring).

```suggestion
        append_mode = mode if "a" in mode else mode + "a"
```